File: 2/2000s/20s/2075/DecodeTheSlantedCipherText.cpp
/*
https://leetcode.com/problems/decode-the-slanted-ciphertext/?envType=daily-question&envId=2026-04-04
*/

class Solution {
public:
    string decodeCiphertext(string encodedText, int rows) {
        int lengthOfText=encodedText.length(), cols=lengthOfText/rows;
        string res="";
        for(int startCol=0; startCol<cols; ++startCol){
            int i=0, j=startCol;
            while(i<rows&&j<cols){
                res+=encodedText[i*cols+j];
                ++i, ++j;
            }
        }
        while(!res.empty()&&res.back()==' '){
            res.pop_back();
        }
        return res;
    }
};

File: 2/2000s/20s/2075/DecodeTheSlantedCipherText.java
/*
https://leetcode.com/problems/decode-the-slanted-ciphertext/?envType=daily-question&envId=2026-04-04
*/

class Solution {
    public String decodeCiphertext(String encodedText, int rows) {
        int lengthOfEncodedText=encodedText.length(), columns=lengthOfEncodedText/rows;
        StringBuilder resultantOriginalString=new StringBuilder();
        for(int startCol=0; startCol<columns; ++startCol){
            int i=0, j=startCol;
            while(i<rows&&j<columns){
                resultantOriginalString.append(encodedText.charAt(i*columns+j));
                ++i;
                ++j;
            }
        }
        while(!resultantOriginalString.isEmpty()&&resultantOriginalString.charAt(resultantOriginalString.length()-1)==' '){
            resultantOriginalString.deleteCharAt(resultantOriginalString.length()-1);
        }
        return String.valueOf(resultantOriginalString);
    }
}